auth/maintenance_socket_role_manager.cc

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (C) 2023-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#include "auth/maintenance_socket_role_manager.hh"
+
+#include <seastar/core/future.hh>
+#include <stdexcept>
+#include <string_view>
+#include "log.hh"
+#include "utils/class_registrator.hh"
+
+namespace auth {
+
+constexpr std::string_view maintenance_socket_role_manager_name = "com.scylladb.auth.MaintenanceSocketRoleManager";
+
+static const class_registrator<
+        role_manager,
+        maintenance_socket_role_manager,
+        cql3::query_processor&,
+        ::service::migration_manager&> registration(sstring{maintenance_socket_role_manager_name});
+
+
+std::string_view maintenance_socket_role_manager::qualified_java_name() const noexcept {
+    return maintenance_socket_role_manager_name;
+}
+
+const resource_set& maintenance_socket_role_manager::protected_resources() const {
+    static const resource_set resources{};
+
+    return resources;
+}
+
+future<> maintenance_socket_role_manager::start() {
+    return make_ready_future<>();
+}
+
+future<> maintenance_socket_role_manager::stop() {
+    return make_ready_future<>();
+}
+
+template<typename T = void>
+future<T> operation_not_supported_exception(std::string_view operation) {
+    return make_exception_future<T>(
+        std::runtime_error(format("role manager: {} operation not supported through maintenance socket", operation)));
+}
+
+future<> maintenance_socket_role_manager::create(std::string_view role_name, const role_config&) {
+    return operation_not_supported_exception("CREATE");
+}
+
+future<> maintenance_socket_role_manager::drop(std::string_view role_name) {
+    return operation_not_supported_exception("DROP");
+}
+
+future<> maintenance_socket_role_manager::alter(std::string_view role_name, const role_config_update&) {
+    return operation_not_supported_exception("ALTER");
+}
+
+future<> maintenance_socket_role_manager::grant(std::string_view grantee_name, std::string_view role_name) {
+    return operation_not_supported_exception("GRANT");
+}
+
```

Comment:
We might decide to allow authentication after all (a useful way to fix broken configuration), but let's leave it like that for now.